python/rikai/spark/sql/codegen/fs.py

```diff
@@ -13,12 +13,121 @@
 #  limitations under the License.
 
 import secrets
-from typing import Dict
+from typing import Any, Dict, Optional, Union, IO, Mapping
 
+import yaml
+from jsonschema import ValidationError, validate
 from pyspark.sql import SparkSession
 
+from rikai.io import open_uri
 from rikai.logging import logger
 from rikai.spark.sql.codegen.base import Registry
+from rikai.spark.sql.exceptions import SpecError
+from rikai.spark.sql.schema import parse_schema
+
+__all__ = ["FileSystemRegistry"]
+
+# YAML-Spec SCHEMA
+SPEC_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "version": {
+            "type": "number",
+            "description": "Model SPEC format version",
+        },
+        "name": {"type": "string", "description": "Model name"},
+        "schema": {"type": "string"},
```

Comment:
i forgot what this schema is?

Reply:
This is used to build the return type of a UDF (pandas_udf) in this case. 

The POC code can be seen here: https://github.com/eto-ai/rikai/pull/119/files#diff-7fccc4b1dbc18bbb3a690fc62930ecddeb86c0f17dc274555a6a4affa61f7e8dR65